File: 2020/day2/part1.rb
input = STDIN.read.split("\n")

valid_amount = 0
regex = /([0-9]+)-([0-9]+) ([a-z]): (.+)/
input.each do |line|
    min, max, letter, password = line.match(regex).captures
    count = password.count letter 
    if (min.to_i <= count) && (count <= max.to_i)
        valid_amount += 1
    end
end

puts valid_amount

File: 2023/day1/part1.ts
const regex = /\d/g;

const input = `input here`;

let sum = 0;

for (const line of input.split("\n")) {
  const nums = line.match(regex)!;
  let ns;
  if (nums.length < 2) ns = nums[0] + "" + nums[0];
  else ns = `${nums[0]}${nums[nums.length - 1]}`;
  const num = Number(ns);
  sum += num;
}
console.log(sum);
